src/App.java

```diff
@@ -1,5 +1,562 @@
+import java.io.FileNotFoundException;
+
 public class App {
+
+    private FileInputOutput io = new FileInputOutput();
+    private Book[] books; //An array of book objects
+    private Library library; //initializing a library object so that you can use library methods 
+    private String[] fileInput; //Adding 
+    private Comms comms= new Comms();
+    private String fileName;
+    private boolean borrowedYN; //Set this variable to true if user responded yes to the question "Is the book borrowed"
+    private int a=5; //Used to add a count down feature in the addBooks() method, needed something that was permanent outside the method, but dynamic with the program
+    private String File1;//Adds an extra file slot so that you can add a file, and effectively have 2 libraries
+     
+ 
+/*Main only runs start, and start runs everything else */
     public static void main(String[] args) throws Exception {
-        System.out.println("Hello, World!");
+        App app= new App();
+        app.start();
     }
+
+
+
+    public void start(){ //calls methods in the order that you want them to happen
+        startInstructions(); //Gives starting instructions
+        fileToBooks();//Converts the added file to books
+        menu(); //Lets users manipulate and change books
+        booksToFile(); //Saves changes back to file to create a permanent program(Program in which changes persist even after the program is closed)
+     }
+    
+       
+
+    public void fileToBooks(){ //All the logic behind how information is read from a file and converted into multiple book objects and a library object
+        int arrSize;//Here I declare my integer value that will be responsible for deciding the size of the array
+        try { //try catch method, checks whether the file actually exists before running the following operations
+            io.openFile(fileName);//opens file
+            arrSize = io.numlines();//saves the number of lines in the file into an array, this ensures that the array isn't bigger than it needs to be
+            io.closeFile();//closes the file, so that when the program reopens the file it reads from the start
+            io.openFile(fileName); // Reopen to reset scanner, and read from the start
+            
+            books = new Book[arrSize+5]; //Makes the array of books as big as arrSize+5, it is +5 to add 5 extra book slots
+            // ... rest of your code ...
+        } catch (FileNotFoundException e) {//catch part of the try catch method
+            comms.consoleOutput("File not found, retry");//prints this if file not found
+            startInstructions();//reruns the start instructions so that user can add a file to that actually exists
+            return;
+        }
+
+        library = new Library(books); //initializes the previously declared library objects, done here as now there is an actual array

```

Comment:
since the `books` array is empty herre, would it have made more sense to let the `Library` constructor actually build the array? You could pass the size as an argument, rather than the array itself